ts/features/bonusVacanze/components/ActiveBonus.tsx

```diff
@@ -0,0 +1,110 @@
+import { Text } from "native-base";
+import * as React from "react";
+import { StyleSheet, View } from "react-native";
+import TouchableDefaultOpacity from "../../../components/TouchableDefaultOpacity";
+import IconFont from "../../../components/ui/IconFont";
+import I18n from "../../../i18n";
+import customVariables from "../../../theme/variables";
+import { formatDateAsLocal } from "../../../utils/dates";
+import { Bonus } from "../mock/mockData";
+
+type Props = {
+  bonus: any;
+  onPress: (bonus: Bonus) => void;
+};
+
+const ICON_WIDTH = 24;
+
+const styles = StyleSheet.create({
+  smallSpacer: {
+    width: "100%",
+    height: 4
+  },
+  spaced: {
+    justifyContent: "space-between",
+    flexDirection: "row",
+    alignItems: "center"
+  },
+  brandDarkGray: {
+    color: customVariables.brandDarkGray
+  },
+  badgeContainer: {
+    flex: 0,
+    paddingRight: 8,
+    alignSelf: "flex-start",
+    paddingTop: 6.5
+  },
+  viewStyle: {
+    flexDirection: "row"
+  },
+  text11: {
+    color: customVariables.brandDarkestGray
+  },
+  text3: {
+    fontSize: 18,
+    color: customVariables.brandDarkestGray
+  },
+  text12: {
+    lineHeight: 18,
+    marginBottom: -4
+  },
+  icon: {
+    width: 64,
+    alignItems: "flex-end",
+    justifyContent: "center"
+  },
+  text3Line: {
+    flex: 1,
+    flexDirection: "row"
+  },
+  text3Container: {
+    flex: 1,
+    flexDirection: "row",
+    minHeight: 24
+  }
+});
+
+/**
+ * Component to display the current active bonus if it is present
+ * in the store
+ */
+class ActiveBonus extends React.PureComponent<Props> {
```

Comment:
Since this is a component stateless, could we write it as a Functional component like you did for **RequestBonus**?

Reply:
In 79fbb42 the requested change 😄